gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/ExecutionVisitor.java

```diff
@@ -146,81 +156,53 @@ public EntityFetcherResponse visit(SelectionNode selectionNode) {
-                        executionContext.getRequestHeaders());
-                return entityFetcher.getEntitiesAndAggregatedMetrics(context, request);
-              })
-          .collect(Collectors.toList()));
-    } else {
-      // if data are coming from multiple sources, then, get entities and aggregated metrics
-      // needs to be separated
-      resultMapList.addAll(selectionNode.getAttrSelectionSources().parallelStream()
-          .map(
-              source -> {
-                EntitiesRequest request =
-                    EntitiesRequest.newBuilder(executionContext.getEntitiesRequest())
-                        .clearSelection()
-                        .clearFilter()
-                        .addAllSelection(
-                            executionContext.getSourceToSelectionExpressionMap().get(source))
-                        .setFilter(filter)
-                        .build();
-                IEntityFetcher entityFetcher = queryHandlerRegistry.getEntityFetcher(source);
-                EntitiesRequestContext context =
-                    new EntitiesRequestContext(
-                        executionContext.getTenantId(),
-                        request.getStartTimeMillis(),
-                        request.getEndTimeMillis(),
-                        request.getEntityType(),
-                        executionContext.getRequestHeaders());
-                return entityFetcher.getEntities(context, request);
-              })
-          .collect(Collectors.toList()));
-      resultMapList.addAll(
-          selectionNode.getAggMetricSelectionSources().parallelStream()
-              .map(
-                  source -> {
-                    EntitiesRequest request =
-                        EntitiesRequest.newBuilder(executionContext.getEntitiesRequest())
-                            .clearSelection()
-                            .clearFilter()
-                            .addAllSelection(
-                                executionContext.getSourceToMetricExpressionMap().get(source))
-                            .setFilter(filter)
-                            .build();
-                    IEntityFetcher entityFetcher = queryHandlerRegistry.getEntityFetcher(source);
-                    EntitiesRequestContext context =
-                        new EntitiesRequestContext(
-                            executionContext.getTenantId(),
-                            request.getStartTimeMillis(),
-                            request.getEndTimeMillis(),
-                            request.getEntityType(),
-                            executionContext.getRequestHeaders());
-                    return entityFetcher.getAggregatedMetrics(context, request);
-                  })
-              .collect(Collectors.toList()));
-    }
+    // if data are coming from multiple sources, then, get entities and aggregated metrics
+    // needs to be separated
+    resultMapList.addAll(selectionNode.getAttrSelectionSources().parallelStream()
```

Comment:
which threadpool does this parallel stream use? If it's the common FJP, these queries might be bottlenecked there.